vegacapsule nodes remove does not remove visor directories

The tests run a command like so:
```
2022-10-12 16:42:27.044:[INFO] [MainProcess] [MainThread] [   commands.py]:[remove_validator]:[  71] /workspace/system-tests/vegacapsule-bin nodes remove --name testnet-nodeset-validators-6-validator --home-path /workspace/system-tests/../networkdata/testnet
2022-10-12 16:42:27.303:[INFO] [MainProcess] [MainThread] [   commands.py]:[remove_validator]:[  77]
```
Which results in the directory under `vega` dir disappearing, but there is still a dir under `visor`, which results in the next test which attempts to add a new node failing because this directory already exists, like so:
```
E       AssertionError: Error adding node. Stderr is:
E         2022/10/12 16:50:30 Initiating Tendermint node "validator" with: /workspace/networkdata/vega [tm init validator --home /workspace/networkdata/testnet/tendermint/node6]
E         2022/10/12 16:50:30 I[2022-10-12|16:50:30.744] Generated private validator                  module=main keyFile=/workspace/networkdata/testnet/tendermint/node6/config/priv_validator_key.json stateFile=/workspace/networkdata/testnet/tendermint/node6/data/priv_validator_state.json
E         I[2022-10-12|16:50:30.745] Generated node key                           module=main path=/workspace/networkdata/testnet/tendermint/node6/config/node_key.json
E         I[2022-10-12|16:50:30.745] Generated genesis file                       module=main path=/workspace/networkdata/testnet/tendermint/node6/config/genesis.json
E
E         2022/10/12 16:50:30 Initiating node "validator" with: /workspace/networkdata/vega [init --home /workspace/networkdata/testnet/vega/node6 --nodewallet-passphrase-file /workspace/networkdata/testnet/vega/node6/node-vega-wallet-pass.txt --no-tendermint --output json validator]
E         2022/10/12 16:50:30 Creating vega wallet with: /workspace/networkdata/vega [wallet --home /workspace/networkdata/testnet/vega/node6 create --output json --passphrase-file /workspace/networkdata/testnet/vega/node6/vega-wallet-pass.txt --wallet created-wallet]
E         2022/10/12 16:50:31 node wallet create out: &vega.createWalletOutput{Wallet:struct { Name string "json:\"name\""; FilePath string "json:\"filePath\""; RecoveryPhrase string "json:\"recoveryPhrase\"" }{Name:"created-wallet", FilePath:"/workspace/networkdata/testnet/vega/node6/data/wallets/created-wallet", RecoveryPhrase:"kitten kit tortoise prevent width leave kind flush involve remain source reject hidden visual apple magnet uphold beef idle surface social tongue extra safe"}, Key:struct { Public string "json:\"publicKey\"" }{Public:"e8960e6531005c7c4d65219aab22da9368b53c1786abd2977dc2f5518a8c1765"}}
E         2022/10/12 16:50:31 Importing node vega wallet with: /workspace/networkdata/vega [nodewallet --home /workspace/networkdata/testnet/vega/node6 --passphrase-file /workspace/networkdata/testnet/vega/node6/node-vega-wallet-pass.txt import --output json --chain vega --wallet-passphrase-file /workspace/networkdata/testnet/vega/node6/vega-wallet-pass.txt --wallet-path /workspace/networkdata/testnet/vega/node6/data/wallets/created-wallet]
E         2022/10/12 16:50:31 node wallet import out: &vega.importNodeWalletOutput{RegistryFilePath:"/workspace/networkdata/testnet/vega/node6/config/node/wallets.encrypted", TendermintPubkey:"", WalletFilePath:"/workspace/networkdata/testnet/vega/node6/data/node/wallets/vega/vega.1665589831398901900"}
E         2022/10/12 16:50:31 Generating node "ethereum" wallet with: /workspace/networkdata/vega [nodewallet --home /workspace/networkdata/testnet/vega/node6 --passphrase-file /workspace/networkdata/testnet/vega/node6/node-vega-wallet-pass.txt generate --output json --chain ethereum --wallet-passphrase-file /workspace/networkdata/testnet/vega/node6/ethereum-vega-wallet-pass.txt]
E         2022/10/12 16:50:39 ethereum wallet out: &vega.generateNodeWalletOutput{Mnemonic:"", RegistryFilePath:"/workspace/networkdata/testnet/vega/node6/config/node/wallets.encrypted", WalletFilePath:"/workspace/networkdata/testnet/vega/node6/data/node/wallets/ethereum/UTC--2022-10-12T15-50-31.624824900Z--987f016820d0fc94b77abe3f0dc9194a1bcddd98"}
E         2022/10/12 16:50:40 Importing tenderming wallet: [nodewallet --home /workspace/networkdata/testnet/vega/node6 --passphrase-file /workspace/networkdata/testnet/vega/node6/node-vega-wallet-pass.txt import --output json --chain tendermint --tendermint-home /workspace/networkdata/testnet/tendermint/node6]
E         2022/10/12 16:50:40 tendermint wallet out: &vega.importNodeWalletOutput{RegistryFilePath:"/workspace/networkdata/testnet/vega/node6/config/node/wallets.encrypted", TendermintPubkey:"jrxEXJ4KzikuSiJFc3frBNLdL+WESCTA6BX2iVKlbMQ=", WalletFilePath:""}
E         2022/10/12 16:50:40 vega config initialized for node "validator" with id 6, paths: "/workspace/networkdata/testnet/vega/node6/config/node/config.toml"
E         2022/10/12 16:50:40 Initiating visor with: /workspace/networkdata/visor [init --home /workspace/networkdata/testnet/visor/visor6]
E         2022/10/12 16:50:40 Trying to stop jobs: clef-3, clef-4, ganache-1, postgres-1, testnet-faucet, testnet-nodeset-full-5-full, testnet-nodeset-validators-0-validator, testnet-nodeset-validators-1-validator, testnet-nodeset-validators-2-validator, testnet-nodeset-validators-clef-3-validator, testnet-nodeset-validators-clef-4-validator, testnet-vegawallet
E         2022/10/12 16:50:40 Job "clef-4" stopped
E         2022/10/12 16:50:40 Job "testnet-nodeset-full-5-full" stopped
E         2022/10/12 16:50:40 Job "testnet-nodeset-validators-2-validator" stopped
E         2022/10/12 16:50:40 Job "testnet-vegawallet" stopped
E         2022/10/12 16:50:40 Job "ganache-1" stopped
E         2022/10/12 16:50:40 Job "testnet-nodeset-validators-clef-4-validator" stopped
E         2022/10/12 16:50:40 Job "testnet-nodeset-validators-clef-3-validator" stopped
E         2022/10/12 16:50:40 Job "testnet-faucet" stopped
E         2022/10/12 16:50:40 Job "testnet-nodeset-validators-1-validator" stopped
E         2022/10/12 16:50:40 Job "clef-3" stopped
E         2022/10/12 16:50:40 Job "postgres-1" stopped
E         2022/10/12 16:50:40 Job "testnet-nodeset-validators-0-validator" stopped
E         2022/10/12 16:50:40 Jobs have been stopped.
E         Error: failed to add new node: failed to initiate Visor id 6 for node set validators: failed to execute binary /workspace/networkdata/visor [init --home /workspace/networkdata/testnet/visor/visor6] with error: Error: home folder "/workspace/networkdata/testnet/visor/visor6" already exists
E         Usage:
E           vegavisor init [flags]
E
E         Flags:
E           -h, --help             help for init
E               --home string      Path to visor home folder to be generated
E               --with-data-node   Determines whether or not data node config should be also generated
E
E         : exit status 1
E         Usage:
E           /workspace/system-tests/vegacapsule-bin nodes add [flags]
E
E         Flags:
E               --base-on string   Name of the node set that the new node set should be based on
E           -h, --help             help for add
E               --start            Allows to configure whether or not the new node set should automatically start (default true)
E
E         Global Flags:
E               --home-path string   Specify the location of network home directory (default "/home/jake/.vegacapsule/testnet")
E
E
E       assert '' != ''
E        +  where '' = CompletedProcess(args=['/workspace/system-tests/vegacapsule-bin', 'nodes', 'add', '--start=false', '--base-on', 'testnet-nodeset-validators-0-validator', '--home-path', '/workspace/system-tests/../networkdata/testnet'], returncode=1, stdout='', stderr='2022/10/12 16:50:30 Initiating Tendermint node "validator" with: /workspace/networkdata/vega [tm init validator --home /workspace/networkdata/testnet/tendermint/node6]\n2022/10/12 16:50:30 I[2022-10-12|16:50:30.744] Generated private validator                  module=main keyFile=/workspace/networkdata/testnet/tendermint/node6/config/priv_validator_key.json stateFile=/workspace/networkdata/testnet/tendermint/node6/data/priv_validator_state.json\nI[2022-10-12|16:50:30.745] Generated node key                           module=main path=/workspace/networkdata/testnet/tendermint/node6/config/node_key.json\nI[2022-10-12|16:50:30.745] Generated genesis file                       module=main path=/workspace/networkdata/testnet/tendermint/node6/config/genesis.json\n\n2022/10/12 16:50:30 Initiating node "validator" with: /workspace/networkdata/vega [init --home /workspace/networkdata/testnet/vega/node6 --nodewallet-passphrase-file /workspace...3" stopped\n2022/10/12 16:50:40 Job "postgres-1" stopped\n2022/10/12 16:50:40 Job "testnet-nodeset-validators-0-validator" stopped\n2022/10/12 16:50:40 Jobs have been stopped.\nError: failed to add new node: failed to initiate Visor id 6 for node set validators: failed to execute binary /workspace/networkdata/visor [init --home /workspace/networkdata/testnet/visor/visor6] with error: Error: home folder "/workspace/networkdata/testnet/visor/visor6" already exists\nUsage:\n  vegavisor init [flags]\n\nFlags:\n  -h, --help             help for init\n      --home string      Path to visor home folder to be generated\n      --with-data-node   Determines whether or not data node config should be also generated\n\n: exit status 1\nUsage:\n  /workspace/system-tests/vegacapsule-bin nodes add [flags]\n\nFlags:\n      --base-on string   Name of the node set that the new node set should be based on\n  -h, --help             help for add\n      --start            Allows to configure whether or not the new node set should automatically start (default true)\n\nGlobal Flags:\n      --home-path string   Specify the location of network home directory (default "/home/jake/.vegacapsule/testnet")\n\n').stdout

modules/vegacapsule/commands.py:60: AssertionError
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -46,6 +46,7 @@
 - [248](https://github.com/vegaprotocol/vegacapsule/pull/248) Remove --no-version-check flag from vega wallet in the importer module
 - [273](https://github.com/vegaprotocol/vegacapsule/pull/273) Remove `clef-address` from `nodewallet` config file
 - [228](https://github.com/vegaprotocol/vegacapsule/pull/228) Network can now be stopped/started after previous start failed
+- [289](https://github.com/vegaprotocol/vegacapsule/pull/289) Fix remove node with Visor
 
 
 

--- a/generator/generator.go
+++ b/generator/generator.go
@@ -213,12 +213,16 @@ func (g *Generator) RemoveNodeSet(ns types.NodeSet) error {
 		return fmt.Errorf("failed to remove Vega directory %q: %w", ns.Vega.HomeDir, err)
 	}
 
-	if ns.DataNode == nil {
-		return nil
+	if ns.Visor != nil {
+		if err := os.RemoveAll(ns.Visor.HomeDir); err != nil {
+			return fmt.Errorf("failed to remove Visor directory %q: %w", ns.Vega.HomeDir, err)
+		}
 	}
 
-	if err := os.RemoveAll(ns.DataNode.HomeDir); err != nil {
-		return fmt.Errorf("failed to remove DataNode directory %q: %w", ns.DataNode.HomeDir, err)
+	if ns.DataNode != nil {
+		if err := os.RemoveAll(ns.DataNode.HomeDir); err != nil {
+			return fmt.Errorf("failed to remove DataNode directory %q: %w", ns.DataNode.HomeDir, err)
+		}
 	}
 
 	return nil

--- a/net_confs/node_set_templates/default/visor_run.tmpl
+++ b/net_confs/node_set_templates/default/visor_run.tmpl
@@ -16,8 +16,8 @@ name = "genesis"
 {{ if .NodeSet.DataNode }}
 [data_node]
     [data_node.binary]
-    path = "data-node"
-    args = ["node",
+    path = "vega"
+    args = ["datanode", "node",
       "--home", "{{ .NodeSet.DataNode.HomeDir }}",
     ]
 {{ end }}

